application/uitests/source/class/uitests/DragAndDropApplication.js

```diff
@@ -0,0 +1,156 @@
+/* ************************************************************************
+
+   Copyright:
+
+   License:
+
+   Authors:
+
+ ************************************************************************ */
+
+/**
+ * This is the main application class of your custom application "dragndrop"
+ * 
+ * @asset(qx/icon/Tango/16/actions/document-print.png)
+ */
+qx.Class.define("uitests.DragAndDropApplication", {
+  extend: qx.application.Standalone,
+
+  members: {
+    main: function() {
+      this.base(arguments);
+
+      if (qx.core.Environment.get("qx.debug")) {
+        qx.log.appender.Native;
+        qx.log.appender.Console;
+      }
+
+      var doc = this.getRoot();
+      var tb = new qx.ui.toolbar.ToolBar();
+      var btn = new qx.ui.toolbar.Button("Click Me", "qx/icon/Tango/16/actions/document-print.png");
+      tb.add(btn);
+      doc.add(tb, { left: 10, top: 10, right: 10 });
+      
+      var root = new qx.ui.container.Composite(new qx.ui.layout.HBox(10));
+      doc.add(root, { left: 10, top: 100, right: 10, bottom: 10 });
+
+      var compSource = new qx.ui.container.Composite(new qx.ui.layout.VBox(6));
+      root.add(compSource);
+      compSource.add(new qx.ui.basic.Label("Simple Source"));
+      var lstSource = new qx.ui.form.List().set({ draggable: true, selectionMode: "multi" });
+      compSource.add(lstSource);
+
+      for (var i = 0; i < 20; i++)
+        lstSource.add(new qx.ui.form.ListItem("Item " + i, "icon/16/places/folder.png"));
+
+      var cbxCancelDragstart = new qx.ui.form.CheckBox("Cancel dragstart").set({ value: false });
+      compSource.add(cbxCancelDragstart);
+      
+      var cbxRejectDragstart = new qx.ui.form.CheckBox("Reject dragstart").set({ value: false });
+      compSource.add(cbxRejectDragstart);
+      
+      lstSource.addListener("dragstart", function(e) {
+        this.debug("lstSource: dragstart");
+        return new qx.Promise((resolve, reject) => {
```

Comment:
This may be a bigger discussion than just this PR... Although I'm greatly looking forward to core code being able to use ES6, and it'll be fine for user code to use ES6 along with the new compiler, we're only deprecating, not removing the generator for 6.0, so users should still be able to run and create tests which use the generator. I think we should continue to avoid ES6 in core code until 7.0.

Reply:
oops, forgot to lint that part...

Reply:
ah, on second thoughts this `uitests` folder is designed to produce multiple applications, each of which can be run as independent test applications; the generator only supports a single application model.  The idea behind this `uitests` folder is to create tests that can easily be operated by selenium/etc when it arrives (which should also be in 6.x, soon after I can answer the outstanding question in https://github.com/qooxdoo/qooxdoo/pull/9426)

Also, we've been moving recently towards integrating the compiler into the qooxdoo core repo (i.e. so that it can ship with qooxdoo alongside the generator), but even if that doesn't make it into the 6.0 release the compiler can still be installed automatically thanks to `npm` and `node` is already a requirement for 5.x

Can we then support the compiler/ES6 in specific places like framework tests  provided that it is not in the framework source code itself?

Reply:
I suppose. But for 6.0, there should be no uses of ES6 in the framework source code, and we should aggressively ensure that.